src/urllib3/connectionpool.py

```diff
@@ -164,6 +165,11 @@ class HTTPConnectionPool(ConnectionPool, RequestMethods):
         A dictionary with proxy headers, should not be used directly,
         instead, see :class:`urllib3.ProxyManager`
 
+    :param idle_timeout:
+       Idle timeout in seconds after which a connection is considered not
+       usable. Useful if connecting via NAT that silently drops connections
+       after inactivity period like `AWS NAT gateway <https://docs.aws.amazon.com/vpc/latest/userguide/nat-gateway-troubleshooting.html#nat-gateway-troubleshooting-timeout>`_.
+
```

Comment:
You may want to add as well: https://learn.microsoft.com/en-us/azure/nat-gateway/faq#what-is-the-idle-timeout-for-a-nat-gateway